feat(cli): add @bomb.sh/tab completions

AmirSa12 opened the pull request:
### Description

This PR introduces tab completion functionality for the vitest cli, improving developer experience by allowing shell(zsh, powershell, fish, bash) autocompletion for commands, options, values, and flags. With this PR, users can navigate available commands, options and values more efficiently and faster with speeding up workflow.

A small video of how this works:

https://github.com/user-attachments/assets/a3e09a22-f262-4166-a76a-e4770c4cdd1a

A small documentation of how CLIs can integrate tab ( in this case, I'm using the CAC adapter )
```ts
import cac from 'cac';
import tab from '@bomb.sh/tab/cac';

const cli = cac('my-cli');

cli
  .command('dev', 'Start dev server')
  .option('--port <port>', 'Specify port')
  .option('--host <host>', 'Specify host');

const completion = await tab(cli);

// custom completions for option values
const devCommand = completion.commands.get('dev');
const portOption = devCommand?.options.get('port');
if (portOption) {
  portOption.handler = (complete) => {
    complete('3000', 'Development port');
    complete('8080', 'Production port');
  };
}

cli.parse();
```



### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.
- [x] Please check [Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to make review process faster. Note that this option is not available for repositories that are owned by Github organizations.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

Aslemammad commented on packages/vitest/src/node/cli/cac.ts:
```diff
@@ -233,7 +235,8 @@ export function parseCLI(argv: string | string[], config: CliParseOptions = {}):
   arrayArgs.unshift('node')
-  let { args, options } = createCLI(config).parse(arrayArgs, {
+  const cli = createCLI(config)
+  let { args, options } = cli.parse(arrayArgs, {
```
this is an unnecessary diff! let's remove it.

AmirSa12 commented:
will make this PR ready for preview after the 0.0.5 release which changes the license and reduces the size of the lib by ~ 70%

AmirSa12 commented:
Hey @sheremet-va ! Thanks for yout review!  I have addressed the feedback points. providing value completions is entirely customizable. If you would like specific options and flags to have autocompletion values, please let me know which ones need what values and I can add them. The current implementation focuses on the command and option completions more than the values of the options.

sheremet-va commented on packages/vitest/src/node/cli/completions.ts:
```diff
@@ -0,0 +1,61 @@
+        complete('firefox', 'Mozilla Firefox')
+        complete('safari', 'Safari')
+        complete('edge', 'Microsoft Edge')
+        complete('chromium', 'Chromium')
```
I would have it as chromium/firefox/safari/chrome/edge because playwright is more popular

sheremet-va commented on packages/vitest/src/node/cli/completions.ts:
```diff
@@ -0,0 +1,61 @@
+        complete('edge', 'Microsoft Edge')
+        complete('chromium', 'Chromium')
+      },
+      'browser.provider'(complete) {
```
This is not possible to configure anymore
